src/api/routes/hero_routes.py

```
@@ -0,0 +1,125 @@
+from flask import Blueprint, request, jsonify , current_app
+from api.extensions import db
+from api.models.Hero import Hero
+from api.schemas.hero_schema import HeroSchema
+from sqlalchemy.exc import SQLAlchemyError
+
+hero_blueprint = Blueprint("hero_blueprint", __name__, url_prefix="/heroes")  # Changed to plural "heroes"
+hero_schema = HeroSchema()
+heroes_schema = HeroSchema(many=True)
+
+
+
+#------FETCH HEROS ---------#
+@hero_blueprint.route('/', methods=['GET'])
+def get_heroes():
+    try:
+        current_app.logger.info(f"Getting all Heros")
+
+        heroes = Hero.query.all()
+        return heroes_schema.jsonify(heroes)
+    except SQLAlchemyError as e:
```

You should implement an application-level error handler to catch all exception and produce a correct 500 error response.

```python
from flask import jsonify
from sqlalchemy.exc import SQLAlchemyError
from marshmallow import ValidationError

@app.errorhandler(SQLAlchemyError)
def handle_sqlalchemy_error(e):
    app.logger.error(f"Database error: {str(e)}")
    return jsonify({"error": "A database error occurred."}), 500

@app.errorhandler(ValidationError)
def handle_validation_error(e):
    return jsonify({"validation_error": e.messages}), 400

@app.errorhandler(404)
def handle_not_found(e):
    return jsonify({"error": str(e)}), 404
```